### PR TITLE
Implements and tests more SVE memory functions

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/detail/function/simd/common/to_logical.hpp [hunk 1]
```diff
@@ -58,7 +59,10 @@ EVE_FORCEINLINE auto
 to_logical(C c, eve::as<T>) noexcept
 {
   using l_t = typename as_logical<T>::type;
-  using i_t = as_integer_t<typename as_logical_t<T>::mask_type>;
+  using i_t = std::conditional_t< (T::size()>=128 && sizeof(element_type_t<T>) == 1)
+                                , as_integer_t<typename as_logical_t<T>::mask_type,unsigned>
+                                , as_integer_t<typename as_logical_t<T>::mask_type>
+                                >;
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/common/to_logical.hpp:65]: What's happening? 256 chars overflow?
If unsigned integer works - why not always use unsigned?

File: include/eve/module/core/regular/impl/simd/arm/sve/store.hpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+namespace eve::detail
+{
+// Regular store
+template<real_scalar_value T, typename N, simd_compatible_ptr<wide<T, N>> Ptr>
+EVE_FORCEINLINE void
+store_(EVE_SUPPORTS(sve_), wide<T, N> v, Ptr p)
+requires(sve_abi<abi_t<T, N>> && !has_store_equivalent<wide<T, N>, Ptr>)
+{
+  auto ptr = unalign(p);
+  if constexpr( N() != expected_cardinal_v<T> ) memcpy(ptr, (T const*)(&v), N::value * sizeof(T));
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/impl/simd/arm/sve/store.hpp:24]: This looks bad. We should use a masked store. Just fwd to keep_first(N{}())

File: include/eve/detail/function/simd/common/to_logical.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/common/to_logical.hpp:65]: x86 unsigned integer comparisons are more costly than signed ones.
This prevents pessimisation.

File: include/eve/module/core/regular/impl/simd/arm/sve/store.hpp (hunk 2 above)
Comment (jfalcou) [reply include/eve/module/core/regular/impl/simd/arm/sve/store.hpp:24]: Will fix

File: include/eve/detail/function/simd/arm/sve/load.hpp [hunk 3]
```diff
@@ -15,28 +15,83 @@
...
+  }
+  else
+  {
+    wide<T, N> loaded;
+
+    if constexpr(!C::is_complete)
+    {
+      auto mask = cond.mask(as<wide<T,N>>{});
+      if constexpr( N() == expected_cardinal_v<T> ) loaded = svld1(mask, ptr);
+      else                                          loaded = svldff1(mask, ptr);
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/sve/load.hpp:41]: Please remind what these 2 are? No failure load?

File: include/eve/detail/function/simd/arm/sve/load.hpp [hunk 4]
```diff
@@ -15,28 +15,83 @@
...
+                      C const& cond,
+                      safe_type const&,
+                      eve::as<logical<wide<T, N>>> const&,
+                      Pointer ptr) noexcept
 requires(dereference_as<logical<T>, Pointer>::value && sve_abi<abi_t<T, N>>)
 {
-  auto block = load(ignore_none, safe, eve::as<wide<T, N>>{}, ptr_cast<T const>(ptr));
+  auto const alt = [&]()
+  {
+    if constexpr( C::has_alternative ) return cond.rebase(cond.alternative.mask());
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/sve/load.hpp:70]: Doesn't rebasr work for no alternative?

File: include/eve/module/core/regular/impl/simd/arm/sve/store.hpp [hunk 5]
```diff
@@ -21,8 +21,10 @@ store_(EVE_SUPPORTS(sve_), wide<T, N> v, Ptr p)
 requires(sve_abi<abi_t<T, N>> && !has_store_equivalent<wide<T, N>, Ptr>)
 {
   auto ptr = unalign(p);
-  if constexpr( N() != expected_cardinal_v<T> ) memcpy(ptr, (T const*)(&v), N::value * sizeof(T));
-  else svst1(sve_true<T>(), ptr, v);
+  if constexpr( N() != expected_cardinal_v<T> )
+    store[keep_first(N::value)](bit_cast(v,as<wide<T>>{}),p);
+  else
+    svst1(sve_true<T>(), ptr, v);
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/impl/simd/arm/sve/store.hpp:27]: Can we do smth wit formatting

File: include/eve/detail/function/simd/arm/sve/load.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/sve/load.hpp:41]: Yes they are

File: include/eve/detail/function/simd/arm/sve/load.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/sve/load.hpp:70]: rebase if the member of the `.else_` subobject so nah.
This should be cleaned up later I guess as this API shoudl eb callable without needing to know about the presence of an alternative.

When called on non-alt conditional, it should just append a .else_()

File: include/eve/module/core/regular/impl/simd/arm/sve/store.hpp (hunk 5 above)
Comment (jfalcou) [reply include/eve/module/core/regular/impl/simd/arm/sve/store.hpp]: Done

File: include/eve/detail/function/simd/arm/sve/load.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/sve/load.hpp:70]: Gonna make a story for that

File: include/eve/detail/function/simd/arm/sve/load.hpp [hunk 6]
```diff
@@ -15,28 +15,73 @@
...
+    else return svldff1(sve_true<T>(), ptr);
+  }
+  else
+  {
+    wide<T, N> loaded;
+
+    if constexpr(!C::is_complete)
+    {
+      auto mask = cond.mask(as<wide<T,N>>{});
+      if constexpr( N() == expected_cardinal_v<T> ) loaded = svld1(mask, ptr);
```
Comment (DenisYaroshevskiy) [inline include/eve/detail/function/simd/arm/sve/load.hpp:40]: This is not correct

Comment (DenisYaroshevskiy) [reply include/eve/detail/function/simd/arm/sve/load.hpp:40]: Ah you used a mask. I thought we are just using the unsafe load

**Review by DenisYaroshevskiy — APPROVED** [review]
I'll approve

**Pull request merged**